api/api/src/main/scala/uk/ac/wellcome/platform/api/services/CollectionService.scala

```diff
@@ -36,13 +35,16 @@ import uk.ac.wellcome.platform.api.Tracing
   * incremental exploration of the data (akin to exploring a filesystem), due to
   * the fact the whole tree may be rather large.
   */
-class CollectionService(elasticClient: ElasticClient)(
+class CollectionService(elasticsearchService: ElasticsearchService)(
   implicit ec: ExecutionContext)
     extends Tracing {
 
   def retrieveTree(index: Index,
                    expandedPaths: List[String]): Future[Result[Collection]] =
-    makeEsRequest(index, expandedPaths)
+    elasticsearchService
+      .executeSearchRequest(
+        CollectionRequestBuilder(index, expandedPaths).request
+      )
```

Comment:
Nice bit of refactoring so that we use the elastic search service here instead of redefining stuff :+1: